lisa/tests/scheduler/load_tracking.py

```diff
@@ -310,7 +315,8 @@ def get_simulated_pelt(self, task, signal_name):
             # executing
             preempted_value=0,
         )
-        df = trace.analysis.load_tracking.df_task_signal(task, signal_name)
+        cpu = self.wlgen_task.children['test']['cpus']
```

Comment:
s/cpu/cpus/ as this is a set, not a single CPU ID

Reply:
The property should be used unless there is a good reason, along with an assert if we can only deal with len(cpus) == 1 (which is the case, as the test currently can only check the signals of one rq at a time). This will make it obvious in the future if a subclass tries to make a workload with more than one CPU. Alternatively to the assert, you can use:
```
cpu, = ...
```
which will unpack a 1-item iterable into `cpu`, or raise a ValueError otherwise.

Reply:
```
diff --git a/lisa/tests/scheduler/load_tracking.py b/lisa/tests/scheduler/load_tracking.py
index e3d5891b138e..966f7d5456d5 100644
--- a/lisa/tests/scheduler/load_tracking.py
+++ b/lisa/tests/scheduler/load_tracking.py
@@ -193,6 +193,17 @@ Time in seconds for util_avg to converge (i.e. ignored time)
         assert len(tasks) == 1
         return tasks[0]
 
+    @property
+    def cpus(self):
+        """
+        All CPUs used by RTapp workload.
+        """
+        return set(itertools.chain.from_iterable(
+            phase['cpus']
+            for task in self.rtapp_profile.values()
+            for phase in task.phases
+        ))
+
     def get_tags(self):
         return {'cpu': f'{self.cpu}@{self.freq}'}
 
@@ -308,7 +319,6 @@ Time in seconds for util_avg to converge (i.e. ignored time)
         logger = self.get_logger()
         trace = self.trace
         task = trace.get_task_id(task)
-        cpus = trace.analysis.tasks.cpus_of_tasks([task])
 
         df_activation = trace.analysis.tasks.df_task_activation(
             task,
@@ -316,8 +326,11 @@ Time in seconds for util_avg to converge (i.e. ignored time)
             # executing
             preempted_value=0,
         )
-        cpu = self.wlgen_task.children['test']['cpus']
-        df = self._get_trace_signal(task, cpu, signal_name)
+
+        cpus = sorted(self.cpus)
+        assert len(cpus) == 1
+        df = self._get_trace_signal(task, cpus, signal_name)
+
         df = df.copy(deep=False)
 
         # Ignore the first activation, as its signals are incorrect
@@ -337,8 +350,7 @@ Time in seconds for util_avg to converge (i.e. ignored time)
         except KeyError:
             if any(
                 self.plat_info['cpu-capacities']['rtapp'][cpu] != UTIL_SCALE
-                for phase in self.wlgen_task.phases
-                for cpu in phase['cpus']
+                for cpu in self.cpus
             ):
                 ResultBundle.raise_skip('PELT time scaling can only be simulated when the PELT clock is available from the trace')
 
@@ -346,6 +358,7 @@ Time in seconds for util_avg to converge (i.e. ignored time)
             clock = None
 
         try:
+            cpus = trace.analysis.tasks.cpus_of_tasks([task])
             capacity = trace.analysis.load_tracking.df_cpus_signal('capacity', cpus)
         except MissingTraceEventError:
             capacity = None
```

Reply:
OK, I'll use:

```
+        cpus = sorted(self.cpus)
+        assert len(cpus) == 1
+        df = self._get_trace_signal(task, cpus, signal_name)
```

and:

```
         try:
+            cpus = trace.analysis.tasks.cpus_of_tasks([task])
             capacity = trace.analysis.load_tracking.df_cpus_signal('capacity', cpus)
```
in def get_simulated_pelt(self, task, signal_name).